### DISCUSSION TITLE
Presto (trino 359) show error Catalog must be specified

### DISCUSSION THREAD

**Discussion by RajatGarga** [cloudera/hue#2397] [Questions]
Is the issue already present in https://github.com/cloudera/hue/issues or discussed in the forum https://discourse.gethue.com?

Describe the bug: When trying to connect to trino 359 I get the following error
```
{
    "message": "line 1: 1: Catalog must be specified when session catalog is not set",
    "errorCode": 56,
    "errorName": "MISSING_CATALOG_NAME",
    "errorType": "USER_ERROR",
    "errorLocation": {
        "lineNumber": 1,
        "columnNumber": 1
    },
    "failureInfo": {
        "type": "io.trino.spi.TrinoException",
        "message": "line 1: 1: Catalog must be specified when session catalog is not set",
        "suppressed": [],
        "stack": [
            "io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java: 48)",
            "io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java: 43)",
            "io.trino.sql.rewrite.ShowQueriesRewrite$Visitor.visitShowSchemas(ShowQueriesRewrite.java: 336)",
            "io.trino.sql.rewrite.ShowQueriesRewrite$Visitor.visitShowSchemas(ShowQueriesRewrite.java: 171)",
            "io.trino.sql.tree.ShowSchemas.accept(ShowSchemas.java: 68)",
            "io.trino.sql.tree.AstVisitor.process(AstVisitor.java: 27)",
            "io.trino.sql.rewrite.ShowQueriesRewrite.rewrite(ShowQueriesRewrite.java: 168)",
            "io.trino.sql.rewrite.StatementRewrite.rewrite(StatementRewrite.java: 61)",
            "io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java: 88)",
            "io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java: 83)",
            "io.trino.execution.SqlQueryExecution.analyze(SqlQueryExecution.java: 269)",
            "io.trino.execution.SqlQueryExecution.(SqlQueryExecution.java: 190)",
            "io.trino.execution.SqlQueryExecution$SqlQueryExecutionFactory.createQueryExecution(SqlQueryExecution.java: 806)",
            "io.trino.dispatcher.LocalDispatchQueryFactory.lambda$createDispatchQuery$0(LocalDispatchQueryFactory.java: 132)",
            "io.trino.$gen.Trino_359____20210726_082910_2.call(Unknown Source)",
            "com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java: 125)",
            "com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java: 69)",
            "com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java: 78)",
            "java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java: 1128)",
            "java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java: 628)",
            "java.base/java.lang.Thread.run(Thread.java: 829)"
        ],
        "errorLocation": {
            "lineNumber": 1,
            "columnNumber": 1
        }
    }
}

```

However I have mentioned catalog in the configuration. I even tried mentioning both catalog and schema with no luck. My configuration is 
```
[[[presto]]]
    name = Trino
    interface=sqlalchemy
    options='{"url": "presto://user:password@XXXX:443/hive-metastore/default","connect_args":"{\"protocol\": \"https\"}"}'
```
Steps to reproduce it? Host hue 4.10.0 via helm and try to connect to trino 359.

Hue version or source? 4.10.1

**Comment by romainr** [comment]
Format looks good https://docs.gethue.com/administrator/configuration/connectors/#trino

Maybe there is a problem with the hyphen '-'. What do you see for `url` here? https://github.com/cloudera/hue/blob/master/desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py#L208 

Does it work if you use the SqlAlchemy code directly, e.g. https://github.com/dropbox/PyHive#sqlalchemy

**Comment by RajatGarga** [comment]
Thanks @romainr! Your comment helped me figure this out.

The hyphen is not the issue since I tried other catalogs that do not have any hyphen. I even got the same error when I used the SqlAlchemy code directly.

This [open issue on PyHive](https://github.com/dropbox/PyHive/issues/378) led me to this [page in Trino's documentation](https://trino.io/blog/2021/01/04/migrating-from-prestosql-to-trino.html) which states that Trino server does not work with older clients and to force older clients to work we have to include the following config property.
```
protocol.v1.alternate-header-name=Presto
```

Adding this configuration property to my Trino server solved this issue. 

Maybe we should have this in the documentation till the driver for Trino is upgraded.

**Comment by romainr** [comment]
Good to know & glad it works!

Would definitely be a good thing to add a note to the docs.

Also maybe adding https://github.com/dungdm93/sqlalchemy-trino for 'trino://' prefix would be clean as it seems like there is no plan to have PyHive work with the latest Trino as of today.

Would you like to add a quick note to the Hue docs about this? https://github.com/cloudera/hue/blob/master/docs/docs-site/content/administrator/configuration/connectors/_index.md

**Comment by RajatGarga** [comment]
Sure! I will make a PR for the docs. 

Yeah we should be using the dialect specific to trino. Maybe I will pick that up over the weekend.
